Sources/Tools/MaxComponent/plGUIComponents.cpp

```diff
@@ -3080,14 +3080,17 @@ class plGUITextBoxProc : public ParamMap2UserDlgProc
                 if ( pmap->GetParamBlock()->GetStr( plGUITextBoxComponent::kRefInitText ) )
                 {
                     fTranslations = plLocalization::StringToLocal(pmap->GetParamBlock()->GetStr( plGUITextBoxComponent::kRefInitText ) );
-                    SetDlgItemText( hWnd, IDC_GUI_INITTEXT, fTranslations[0].c_str() );
+                    ST::wchar_buffer buf = fTranslations[0].to_wchar();
+                    SetDlgItemTextW(hWnd, IDC_GUI_INITTEXT, buf.c_str());
                 }
                 else
                     // if there is no text, then there is nothing to translate
                     SetDlgItemText( hWnd, IDC_GUI_INITTEXT, pmap->GetParamBlock()->GetStr( plGUITextBoxComponent::kRefInitText ) );
                 SendMessage( GetDlgItem( hWnd, IDC_GUI_LANGUAGE ), CB_RESETCONTENT, 0, 0 );
-                for (i=0; i<plLocalization::kNumLanguages; i++)
-                    SendMessageA( GetDlgItem( hWnd, IDC_GUI_LANGUAGE ), CB_ADDSTRING, 0, (LPARAM)plLocalization::GetLanguageName((plLocalization::Language)i) );
+                for (i=0; i<plLocalization::kNumLanguages; i++) {
+                    ST::wchar_buffer languageNameBuf = plLocalization::GetLanguageName((plLocalization::Language)i).to_wchar();
+                    SendMessageW(GetDlgItem(hWnd, IDC_GUI_LANGUAGE), CB_ADDSTRING, 0, (LPARAM)languageNameBuf.c_str());
```

Comment:
I'm not entirely sure this is safe...  If the message is processed right away (as part of the SendMessage call), it should be fine, but if it's processed asynchronously, the buffer may be gone by the time it's used.  Unfortunately, I have no way to test it right now either :(

Reply:
Hrm... I don't have *that* much experience with Win32, so I'm not super familiar with the expected lifetimes for arguments passed via pointers. I would have expected `SendMessage` to make its own copy of the string if it needs to access it later - otherwise I'm not sure how we're supposed to know when the string is safe to deallocate...

Reply:
Okay, according to [the docs](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-sendmessagew), `SendMessageW` "does not return until the window procedure has processed the message", so this *should* be safe.